plugins/woocommerce-blocks/assets/js/data/cart/actions.ts

```diff
@@ -37,9 +38,24 @@ export * from './thunks';
  * @param cart the parsed cart object. (Parsed into camelCase).
  */
 export const setCartData = ( cart: Cart ): { type: string; response: Cart } => {
+	const shippingAddress = applyCheckoutFilter( {
+		filterName: 'setFixedShippingAddress',
```

Comment:
@tarunvijwani  Hi 👋🏼  Can you provide more info on what this filter is for?

I see it's in the `setCartData` action, which looks like it only happens on the `receiveCart` thunk.

Did you add this as a response to my review, specifically this part:

> Also, I'm wondering if we should make sure the initial data of the cart data store contains the readonly address

Also, since you're applying it to both the shipping and billing addresses, the callback function doesn't know the context, you should pass an additional argument specifying the address type.

Thanks!

Reply:
Hi Thomas! The PR is not ready for review yet. I just pushed my local changes before I went to bed last night. 

The filter I added at https://github.com/woocommerce/woocommerce/pull/44416/files#diff-0e9de7819ab0da8d433f06151386a57716b26b359b424c0e8de8f889ddbd670aR457-R463 was not getting called so I was testing with console log whether it will be called here. I'm going to remove this code later. 

>the callback function doesn't know the context, you should pass an additional argument specifying the address type.

That's a great suggestion. I will use the context to set the fixed address.

Reply:
Ok no worries, I'm just wondering why we added this, I don't think it's the right approach if it's in response to this part of my last review:

> Also, I'm wondering if we should make sure the initial data of the cart data store contains the readonly address

we should be ensuring that the cart data store is loaded in with the forced value, setting that via filter isn't a good solution because it means the developer is now setting it once in PHP and once in a JS filter. Moreover it's still possible to update the address using the `setBilling/ShippingAddress` actions without passing the new values through this filter.

Reply:
Marking this conversation resolved as per our Slack discussion.